ldclient/__init__.py

```diff
@@ -12,38 +12,87 @@
 __BUILTINS__ = ["key", "ip", "country", "email",
                 "firstName", "lastName", "avatar", "name", "anonymous"]
 
-
 """Settings."""
-client = None
-sdk_key = None
 start_wait = 5
-config = Config()
 
-_lock = ReadWriteLock()
+__client = None
+__config = Config()
+__lock = ReadWriteLock()
 
 
+# sets config.
+def set_config(config):
+    global __config
+    global __client
+    global __lock
+    try:
+        __lock.lock()
+        if __client:
+            log.info("Reinitializing LaunchDarkly Client " + version.VERSION + " with new config")
+            new_client = LDClient(config, start_wait)
+            old_client = __client
+            __client = new_client
+            old_client.close()
+    finally:
+        __config = config
+        __lock.unlock()
+
+
+# 2 use cases:
+# 1. Initial setup: sets the sdk key for the uninitialized client
+# 2. Allows on-the-fly changing of the sdk key. When this function is called after the client has been initialized
+#    the client will get re-initialized with the new sdk key. In order for this to work, the return value of
```

Comment:
This is the pitfall of this approach, but I don't see another way to do it. any ideas @pkaeding ?